ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -106,7 +107,7 @@
     this.factory = factory;
     this.diskIOScheduler = diskIOScheduler;
     this.time = time;
-    scanSizeInBytes = Math.min(config.storeCleanupOperationsBytesPerSec * 10, 1024 * 1024);
+    scanSizeInBytes = config.storeHardDeleteOperationsBytesPerSec;
```

Comment:
what's this change for?

Reply:
`Math.min` enforces the max value to be 1MB, but people may want to have more than 1MB as hardware upgrade.

Reply:
I _think_ the `* 10` is because `PersistentIndex::findDeletedEntriesSince()` counts the size of _all_ blobs and not just the deleted blobs that it is going to return. The `* 10` was a compensation for that behavior. Is that no longer needed? (maybe not since hard delete loops around anyway but want to make sure it wasn't some kind of optimization that we are rolling back now). cc @pnarayanan

Reply:
this was being capped at `1024 * 1024` because of the `Math.min()`, so the suggestion was to instead enforce things within `StoreConfig`. The value being used would therefore be either `config * 10` or `1024*1024`, so the default for the config should be the one that is being used in practice (which I think @zzmao believes is the latter).

Reply:
`scanSizeInBytes = 10 * config.storeHardDeleteOperationsBytesPerSec;`
 and default `config.storeHardDeleteOperationsBytesPerSec` is 100KB.